### PR TITLE
Rename loop enums

### PR DESCRIPTION
**Pull request by pvuorela**
@inzanity @jpetrell

### CODE REVIEW

File: src/mpris.h [hunk 1]
```diff
@@ -44,9 +44,12 @@ class AMBER_MPRIS_EXPORT Mpris
...
     enum LoopStatus {
-        None,
-        Track,
-        Playlist
+        LoopNone,
+        LoopTrack,
+        LoopPlaylist,
+        None = LoopNone, // this and the following deprecated
+        Track = LoopTrack,
+        Playlist = LoopPlaylist
```
Comment (inzanity) [inline src/mpris.h:52]: These compatibility values probably keep `MprisPlayerPrivate::setLoopStatus()` afloat, but `MprisPlayerPrivate::loopStatus` probably starts to spew out the wrong strings.

Comment (pvuorela) [reply src/mpris.h:52]: Separated the d-bus strings and the corresponding enums there.

File: src/mprisplayer.cpp [hunk 2]
```diff
@@ -93,21 +93,38 @@ qlonglong MprisPlayerPrivate::position() const
...
+    }
 }
 
 QString MprisPlayerPrivate::loopStatus() const
 {
-    const char *strVal = QMetaEnum::fromType<Mpris::LoopStatus>().valueToKey(static_cast<int>(parent()->loopStatus()));
-    return QString::fromLatin1(strVal);
+    switch (parent()->loopStatus()) {
+    case Mpris::LoopNone:
+        return QLatin1String("None");
```
Comment (inzanity) [inline src/mprisplayer.cpp:120]: Wouldn't QStringLiteral be more efficient here, as the value is used to construct a QString?

Comment (pvuorela) [reply src/mprisplayer.cpp:120]: Well, AFAIU the compiler wouldn't be then able to reuse the c string literals that are now needed on both sides of the mapping. That problem could be avoided by global strings for these, but as this is hardly a hot code path I'm not sure if that's worth it.

Comment (inzanity) [reply src/mprisplayer.cpp:120]: Yeah, probably not that important.

File: src/mprisplayer.cpp [hunk 3]
```diff
@@ -93,21 +93,38 @@ qlonglong MprisPlayerPrivate::position() const
 
 void MprisPlayerPrivate::setLoopStatus(const QString &value)
 {
-    bool ok;
-    int enumVal = QMetaEnum::fromType<Mpris::LoopStatus>().keyToValue(value.toUtf8(), &ok);
+    int enumVal;
+    bool ok = true;
```
Comment (inzanity) [inline src/mprisplayer.cpp:97]: I would prefer to skip the `ok`, and just `sendErrorReply(); return`  in else, but either way works.

**Review by inzanity — APPROVED** [review]

File: src/mprisplayer.cpp (hunk 3 above)
Comment (pvuorela) [reply src/mprisplayer.cpp:97]: Both having good sides, this one avoiding return in the middle.

**Pull request merged**